packages/issuer/lib/api.ts

```diff
@@ -0,0 +1,137 @@
+    .withAuthorizationServer(process.env.authorization_server as string)
+    .withCredentialEndpoint(process.env.credential_endpoint as string)
+    .withCredentialIssuer(process.env.credential_issuer as string)
+    .withIssuerDisplay({
+      name: process.env.issuer_name as string,
+      locale: process.env.issuer_locale as string,
+    })
+    .withCredentialsSupported(credentialsSupported)
+    .build()
+}
+
+export class RestAPI {
+  public express: Express
+  private _vcIssuer: VcIssuer
+  //fixme: use this map for now as an internal mechanism for preAuthorizedCode to ids
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  private tokenToId: Map<string, string> = new Map()
+
+  constructor() {
+    dotenv.config()
+    this._vcIssuer = buildVcIssuer()
+    this.express = express()
+    const port = process.env.PORT || 3000
+    const secret = process.env.COOKIE_SIGNING_KEY
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    this.express.use((req, res, next) => {
+      res.header('Access-Control-Allow-Origin', '*')
+      // Request methods you wish to allow
+      res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS, PUT, PATCH, DELETE')
+
+      // Request headers you wish to allow
+      res.setHeader('Access-Control-Allow-Headers', 'X-Requested-With,content-type')
+
+      // Set to true if you need the website to include cookies in the requests sent
+      // to the API (e.g. in case you use sessions)
+      res.setHeader('Access-Control-Allow-Credentials', 'true')
+      next()
+    })
+    this.express.use(bodyParser.urlencoded({ extended: true }))
+    this.express.use(bodyParser.json())
+    this.express.use(cookieParser(secret))
+    this.express.listen(port as number, '0.0.0.0', () => console.log(`Listening on port ${port}`))
+    this.registerMetadataEndpoint()
+    this.registerTokenRequestEndpoint()
+    this.registerCredentialRequestEndpoint()
+    this.registerCredentialOfferEndpoint()
+  }
+
+  private static sendErrorResponse(response: Response, statusCode: number, message: string) {
+    response.statusCode = statusCode
+    response.status(statusCode).send(message)
+  }
+
+  private registerMetadataEndpoint() {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    this.express.get('/metadata', (request, response) => {
+      return response.send(this._vcIssuer._issuerMetadata)
+    })
+  }
+  private registerTokenRequestEndpoint() {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    this.express.post('/token', (request, response) => {
+      return RestAPI.sendErrorResponse(response, 501, 'Not implemented')
+    })
+  }
+  private registerCredentialRequestEndpoint() {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    this.express.get('/credential-request', async (request, response) => {
```

Comment:
Why is this a GET? Doesn't our client lib POST to the credentialOffer endpoint and doesn't the spec say POST?https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#section-7.2

Reply:
changed this to `post`